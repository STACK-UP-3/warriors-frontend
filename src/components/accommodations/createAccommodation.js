import React from 'react';
import Navbar from '../navBar';
import SideMenu from '../common/sideMenu';
import MobileSidebar from '../common/mobileSidebar';
import Footer from '../footer';
import CreateAccomForm from './accommodationForm';

export default () => (
  <div>
    <Navbar />
    <main className='createAccomContainer'>
      <div className='sideMenu'>
        <div className='fixed-sidebar container center hide-on-med-and-down'>
          <SideMenu />
        </div>
        <div className='show-on-med-and-down container center'>
          <MobileSidebar />
        </div>
      </div>
      <div className='mainContent'>
        <div className='row'>
          <div className='col s12 m2'></div>
          <div className='col s12 m10 content'>
            <h5 className='center'>Create Accommodation</h5>
            <CreateAccomForm />
          </div>
        </div>
      </div>
    </main>
    <Footer />
  </div>
);
