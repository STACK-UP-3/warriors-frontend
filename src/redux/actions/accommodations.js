import { CREATE_ACCOMMODATION } from './actionTypes';
import httpHandle from '../../helpers/fetchHandle';

export const createAccommodationStarted = () => ({
  type: CREATE_ACCOMMODATION,
  loading: true,
});

export const createAccommodationSucceded = (payload) => ({
  type: CREATE_ACCOMMODATION,
  loading: false,
  completed: true,
  error: false,
  payload,
});

export const createAccommodationFailed = (payload) => ({
  type: CREATE_ACCOMMODATION,
  loading: false,
  error: true,
  message: payload.message,
});

export const oncreateAccommodation = (accommodationData) => async dispatch => {
  try {
    dispatch(createAccommodationStarted());
    const res = await httpHandle.post('/accommodations', accommodationData);
    /* istanbul ignore next */
    dispatch(createAccommodationSucceded(res.data));
  } catch (error) {
    /* istanbul ignore next */
    const response = {
      error: true,
      message: error.response
        ? error.response.data.message
        : 'Failed connecting to the server, please try again',
    };
    /* istanbul ignore next */
    dispatch(createAccommodationFailed(response));
  }
};
