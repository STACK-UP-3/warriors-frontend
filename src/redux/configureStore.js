import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
