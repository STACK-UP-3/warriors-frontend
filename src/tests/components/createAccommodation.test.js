import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import mockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CreateAccomodationComponent from '../../components/accommodations/createAccommodation';
import CreateAccomodationFormDefault, {
  AddButton,
} from '../../components/accommodations/accommodationForm';
import { AccommodationForm } from '../../components/accommodations/accommodationForm';
import configureStore from '../../redux/configureStore';
import { RemoveButton } from '../../components/accommodations/accommodationForm';

describe('Test <create accommodation/> Component', () => {
  it('Should render component correctly', () => {
    const accomodatioComponent = shallow(<CreateAccomodationComponent />);
    expect(toJson(accomodatioComponent)).toMatchSnapshot();
    expect(accomodatioComponent.length).toBe(1);
  });
});

describe('Test <create Accommodation form/> Component', () => {
  const store = configureStore();
  let handleSubmit;
  beforeEach(() => {
    handleSubmit = jest.fn();
  });

  it('Should render the create accommodation form correctly', () => {
    const store = mockStore([thunk])({
      accomCreateResponse: {
        loading: false,
        error: false,
        message: '',
      },
      cities: [],
    });

    const wrapper = mount(
      <Provider store={store}>
        <CreateAccomodationFormDefault />
      </Provider>,
    );

    const handleSubmit = jest.spyOn(
      wrapper.find('[testdata="createAccommodationForm"]').props(),
      'onSubmit',
    );

    wrapper.find('[testdata="createAccommodationForm"]').props().onSubmit();
    expect(wrapper).toMatchSnapshot();
    expect(handleSubmit).toHaveBeenLastCalledWith();
  });

  it('Should render add and remove buttons', () => {
    const wraper = mount(<RemoveButton remove={jest.fn()} index={0} />);
    const wraper2 = mount(<AddButton add={jest.fn()} item='' />);
    wraper.find('button').at(0).simulate('click');
    wraper2.find('button').at(0).simulate('click');
    expect(wraper.length).toBe(1);
    expect(wraper2.length).toBe(1);
  });

  it('Should test form inputs on change', () => {
    const wrapper = mount(
      <Provider store={store}>
        <CreateAccomodationFormDefault />
      </Provider>,
    );

    wrapper
      .find('input')
      .at(0)
      .simulate('change', {
        target: { value: 'name', name: 'name' },
      });
    wrapper
      .find('input')
      .at(1)
      .simulate('change', {
        target: { value: 'owner', name: 'owner' },
      });
    wrapper
      .find('input')
      .at(2)
      .simulate('change', {
        target: { value: 'additionals', name: 'description' },
      });
    wrapper
      .find('input')
      .at(3)
      .simulate('change', {
        target: { value: 'additionals', name: 'name' },
      });
    wrapper
      .find('input')
      .at(4)
      .simulate('change', {
        target: { value: 'additionals', name: 'cost' },
      });
    wrapper
      .find('input')
      .at(5)
      .simulate('change', {
        target: { value: 'additionals', name: 'details' },
      });
    wrapper
      .find('input')
      .at(6)
      .simulate('change', {
        target: { value: 'additionals' },
      });
    wrapper
      .find('input')
      .at(7)
      .simulate('change', {
        target: { value: 'additionals' },
      });
    wrapper
      .find('input')
      .at(8)
      .simulate('change', {
        target: { value: 'additionals', name: 'details' },
      });
    wrapper
      .find('input')
      .at(9)
      .simulate('change', {
        target: { value: 'additionals' },
      });
    wrapper
      .find('input')
      .at(10)
      .simulate('change', {
        target: { value: 'additionals' },
      });
    wrapper
      .find('input')
      .at(11)
      .simulate('change', {
        target: { value: 'additionals', name: 'name' },
      });
    wrapper
      .find('input')
      .at(12)
      .simulate('change', {
        target: { value: 'additionals', name: 'roomNumber' },
      });
    wrapper
      .find('input')
      .at(13)
      .simulate('change', {
        target: { value: 'additionals', name: 'cost' },
      });
    wrapper
      .find('input')
      .at(14)
      .simulate('change', {
        target: { value: 'additionals', name: 'similarRooms' },
      });
    wrapper
      .find('input')
      .at(15)
      .simulate('change', {
        target: { value: 'additionals', name: 'similarRooms' },
      });
    wrapper
      .find('select')
      .at(0)
      .simulate('change', {
        target: { value: 'location', name: 'location' },
      });
    wrapper
      .find('select')
      .at(1)
      .simulate('change', {
        target: { value: 'type', name: 'type' },
      });
    wrapper
      .find('select')
      .at(2)
      .simulate('change', {
        target: { value: 'status', name: 'status' },
      });
    wrapper
      .find('select')
      .at(3)
      .simulate('change', {
        target: { value: 'status', name: 'type' },
      });
    wrapper
      .find('select')
      .at(4)
      .simulate('change', {
        target: { value: 'status', name: 'status' },
      });
    expect(wrapper.length).toBe(1);
  });

  it('Should test buttons on click', () => {
    const wrapper = mount(
      <Provider store={store}>
        <CreateAccomodationFormDefault />
      </Provider>,
    );

    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    wrapper.find('button').at(3).simulate('click');
    expect(wrapper.length).toBe(1);
  });

  it('Should handle form submit process', () => {
    const form = shallow(
      <AccommodationForm
        citiesProcess={jest.fn()}
        accommodationProcess={jest.fn()}
        cities={{
          data: [
            { id: 1, city: 'kigali' },
            { id: 1, city: 'bujumbura' },
          ],
        }}
        accomCreateResponse={{ loading: true, error: false }}
      />,
    );
    const env = {
      preventDefault: jest.fn(),
    };
    const instance = form.instance();
    const data = {
      name: 'kigali',
      location: 'bujumbura',
      type: 'VIP',
      status: '',
      description: '',
      owner: '',
      services: [],
      images: [],
      rooms: [],
    };
    instance.setState({ ...data });
    instance.handleSubmit(env);
    expect(instance.state).toEqual(data);
  });

  it('Should handle form methods process', () => {
    const form = shallow(
      <AccommodationForm
        citiesProcess={jest.fn()}
        accommodationProcess={jest.fn()}
        cities={{
          data: [
            { id: 1, city: 'kigali' },
            { id: 1, city: 'bujumbura' },
          ],
        }}
        accomCreateResponse={{ loading: false, error: false }}
      />,
    );
    const instance = form.instance();
    instance.removeItem(null, 'images', 0);
    instance.removeItem(null, 'services', 0);
    instance.addItem(null, 'services');
    instance.onRoomCaptionInputChange({ target: { value: '' } }, 0);
    instance.onItemsInputChange(
      { target: { value: '', name: 'name' } },
      'services',
      0,
    );
    instance.onItemsInputChange(
      { target: { value: '', name: 'name' } },
      'rooms',
      0,
    );
    instance.setState({ images: [{}] });
    instance.onItemsInputChange(
      { target: { value: '', name: 'details' } },
      'images',
      0,
    );
    instance.removeItem(null, 'rooms', 0);
    instance.onImageUpload({ target: { files: [] } }, 'images', 0);
    instance.onImageUpload({ target: { files: [] } }, 'rooms', 0);
    expect(instance.state.services.length).toEqual(1);
    instance.setState({
      services: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    });
    expect(instance.state.services.length).toEqual(13);
    instance.setState({ images: [{ url: 'google.com' }] });
    expect(instance.state.images.length).toBe(1);
    instance.setState({ images: [{}, {}] });
    expect(instance.state.images.length).toBe(2);
    instance.setState({ images: [{}, {}, {}, {}, {}, {}] });
    expect(instance.state.images.length).toBe(6);
    instance.setState({ rooms: [{ image: { url: 'url.com' } }] });
    expect(instance.state.rooms.length).toBe(1);
    instance.setState({ error: true });
    expect(instance.state.error).toBe(true);

    const form4 = shallow(
      <AccommodationForm
        citiesProcess={jest.fn()}
        accommodationProcess={jest.fn()}
        cities={{
          data: [
            { id: 1, city: 'kigali' },
            { id: 1, city: 'bujumbura' },
          ],
        }}
        accomCreateResponse={{ loading: true, error: true }}
      />,
    );
    expect(form4).toMatchSnapshot();
  });
});
