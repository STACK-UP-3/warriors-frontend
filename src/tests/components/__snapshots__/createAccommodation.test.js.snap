// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <create Accommodation form/> Component Should handle form methods process 1`] = `
<div>
  <form
    id="createAccommodationForm"
    onSubmit={[Function]}
    testdata="createAccommodationForm"
  >
    <div
      className="form-content"
    >
      <div
        className="row"
        id="textInput"
      >
        <div
          className="input-field col s12 m6"
        >
          <input
            className="validate"
            disabled={true}
            id="name"
            name="name"
            onChange={[Function]}
            required={true}
            testdata="inputChange"
            type="text"
          />
          <label
            htmlFor="name"
          >
            Name
          </label>
        </div>
        <div
          className="input-field col s12 m6"
          id="onwerField"
        >
          <input
            className="validate"
            disabled={true}
            id="owner"
            name="owner"
            onChange={[Function]}
            required={true}
            testdata="inputChange"
            type="text"
          />
          <label
            htmlFor="owner"
          >
            Owner
          </label>
        </div>
      </div>
      <div
        className="row"
        id="textInput"
      >
        <div
          className="input-field col s12 m6"
          style={
            Object {
              "marginTop": "20px",
            }
          }
        >
          <select
            className="validate"
            disabled={true}
            name="location"
            onChange={[Function]}
            required={true}
          >
            <option
              defaultValue="none"
            >
              Choose Location
            </option>
            <option
              key="1"
              value="kigali"
            >
              kigali
            </option>
            <option
              key="1"
              value="bujumbura"
            >
              bujumbura
            </option>
          </select>
          <label
            htmlFor="location"
          >
            Location
          </label>
        </div>
        <div
          className="input-field col s12 m6"
          style={
            Object {
              "marginTop": "20px",
            }
          }
        >
          <select
            className="validate"
            disabled={true}
            name="type"
            onChange={[Function]}
            required={true}
          >
            <option
              defaultValue="none"
            >
              Choose type
            </option>
            <option
              value="Mansion"
            >
              Mansion
            </option>
            <option
              value="Hotel"
            >
              Hotel
            </option>
            <option
              value="Guest House"
            >
              Guest House
            </option>
          </select>
          <label
            htmlFor="type"
          >
            Type
          </label>
        </div>
      </div>
      <div
        className="row"
        id="textInput"
      >
        <div
          className="input-field col s12 m6"
          style={
            Object {
              "marginTop": "20px",
            }
          }
        >
          <select
            className="validate"
            disabled={true}
            name="status"
            onChange={[Function]}
          >
            <option
              defaultValue="none"
            >
              Choose status
            </option>
            <option
              value="available"
            >
              Available
            </option>
            <option
              value="booked"
            >
              Booked
            </option>
          </select>
          <label
            htmlFor="status"
          >
            Status
          </label>
        </div>
        <div
          className="input-field col s12 m6"
          style={
            Object {
              "marginTop": "20px",
            }
          }
        >
          <input
            className="validate"
            disabled={true}
            id="description"
            name="description"
            onChange={[Function]}
            testdata="inputChange"
            type="text"
          />
          <label
            htmlFor="description"
          >
            Additional Info
          </label>
        </div>
      </div>
      <div
        className="row services"
      >
        <h6
          className="headings"
        >
          Services
        </h6>
        <div
          className="row card service"
          id="textInput"
          key="0"
        >
          <div
            className="input-field col s12 m4"
            style={
              Object {
                "marginTop": 20,
              }
            }
          >
            <input
              className="validate"
              disabled={true}
              id="serviceName0"
              name="name"
              onChange={[Function]}
              required={true}
              testdata="inputChange"
              type="text"
            />
            <label
              htmlFor="serviceName0"
            >
              Sevice Name
            </label>
          </div>
          <div
            className="input-field col s12 m4"
            style={
              Object {
                "marginTop": 20,
              }
            }
          >
            <input
              className="validate"
              disabled={true}
              id="serviceCost0"
              name="cost"
              onChange={[Function]}
              testdata="inputChange"
              type="text"
            />
            <label
              htmlFor="serviceCost0"
            >
              Sevice Cost
            </label>
          </div>
          <div
            className="input-field col s12 m4"
            style={
              Object {
                "marginTop": 20,
              }
            }
          >
            <input
              className="validate"
              disabled={true}
              id="serviceDetails0"
              name="description"
              onChange={[Function]}
              testdata="inputChange"
              type="text"
            />
            <label
              htmlFor="serviceDetails0"
            >
              Sevice Details
            </label>
          </div>
        </div>
        <AddButton
          add={[Function]}
          item="services"
        />
      </div>
      <div
        className="row card accImages"
        id="textInput"
      >
        <h6
          className="headings"
        >
          Accommodation Images
        </h6>
        <div
          className="row"
          key="0"
        >
          <div
            className="file-field input-field col m6 s12"
            style={
              Object {
                "marginTop": "20px",
              }
            }
          >
            <div
              className="btn"
            >
              <input
                accept="image/*"
                onChange={[Function]}
                required={true}
                type="file"
              />
              <i
                className="material-icons"
              >
                folder
              </i>
            </div>
            <div
              className="file-path-wrapper"
            >
              <input
                className="file-path validate"
                placeholder="Upload image"
                type="text"
              />
            </div>
          </div>
          <div
            className="input-field col s12 m6"
            style={
              Object {
                "marginTop": "20px",
              }
            }
          >
            <input
              className="validate"
              disabled={true}
              id="images0"
              name="details"
              onChange={[Function]}
              testdata="inputChange"
              type="text"
            />
            <label
              htmlFor="images0"
            >
              Image Caption
            </label>
          </div>
        </div>
        <AddButton
          add={[Function]}
          item="images"
        />
      </div>
      <div
        className="row rooms"
        id="textInput"
      >
        <h6
          className="headings"
        >
          Rooms
        </h6>
        <div
          className="row card room"
          key="0"
        >
          <div
            className="col s12 m8 image-part"
          >
            <div
              className="file-field input-field"
            >
              <div
                className="btn"
              >
                <input
                  accept="image/*"
                  onChange={[Function]}
                  required={true}
                  type="file"
                />
                <i
                  className="material-icons"
                >
                  folder
                </i>
              </div>
              <div
                className="file-path-wrapper"
              >
                <input
                  className="file-path validate"
                  placeholder="Upload image"
                  type="text"
                />
              </div>
            </div>
            <div
              className="input-field col s12 "
              style={
                Object {
                  "marginTop": 30,
                }
              }
            >
              <input
                className="validate"
                disabled={true}
                id="image0"
                name="details"
                onChange={[Function]}
                testdata="inputChange"
                type="text"
              />
              <label
                htmlFor="image0"
              >
                Image Caption
              </label>
            </div>
          </div>
          <div
            className="col s12 m4"
          >
            <div
              className="row"
            >
              <div
                className="input-field col s12 "
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <input
                  className="validate"
                  disabled={true}
                  id="name0"
                  name="name"
                  onChange={[Function]}
                  required={true}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="name0"
                >
                  Room Name
                </label>
              </div>
              <div
                className="input-field col s12 "
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <input
                  className="validate"
                  disabled={true}
                  id="roomNumber0"
                  name="roomNumber"
                  onChange={[Function]}
                  required={true}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="roomNumber0"
                >
                  Room Number
                </label>
              </div>
              <div
                className="input-field col s12"
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <select
                  className="validate"
                  disabled={true}
                  id="type0"
                  name="type"
                  onChange={[Function]}
                  required={true}
                >
                  <option
                    defaultValue="none"
                  >
                    Choose type
                  </option>
                  <option
                    value="vip"
                  >
                    VIP
                  </option>
                  <option
                    value="medium"
                  >
                    Medium
                  </option>
                  <option
                    value="low level"
                  >
                    Low level
                  </option>
                </select>
                <label
                  htmlFor="type0"
                >
                  Type
                </label>
              </div>
              <div
                className="input-field col s12 "
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <input
                  className="validate"
                  disabled={true}
                  id="cost0"
                  name="cost"
                  onChange={[Function]}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="cost0"
                >
                  Room Cost
                </label>
              </div>
              <div
                className="input-field col s12 "
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <input
                  className="validate"
                  disabled={true}
                  id="similarRooms0"
                  name="similarRooms"
                  onChange={[Function]}
                  required={true}
                  testdata="inputChange"
                  type="number"
                />
                <label
                  htmlFor="similarRooms0"
                >
                  Similar Rooms
                </label>
              </div>
              <div
                className="input-field col s12"
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <select
                  className="validate"
                  disabled={true}
                  id="roomStatus0"
                  name="status"
                  onChange={[Function]}
                >
                  <option
                    defaultValue="none"
                  >
                    Choose status
                  </option>
                  <option
                    value="available"
                  >
                    Available
                  </option>
                  <option
                    value="booked"
                  >
                    Booked
                  </option>
                </select>
                <label
                  htmlFor="roomStatus0"
                >
                  Status
                </label>
              </div>
            </div>
          </div>
        </div>
        <AddButton
          add={[Function]}
          item="rooms"
        />
      </div>
      <div
        className="contentFooter"
      >
        <p
          className="error"
        />
      </div>
    </div>
    <div
      className="form-footer"
    >
      <div
        className="submitBtn"
      >
        <button
          className="btn waves-effect waves-light"
          disabled={true}
          id="submit"
          type="submit"
        >
          <Loader
            color="#fff"
            css=""
            loading={true}
            margin={5}
            size={5}
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Test <create Accommodation form/> Component Should render the create accommodation form correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AccommodationForm)>
    <AccommodationForm
      accomCreateResponse={
        Object {
          "error": false,
          "loading": false,
          "message": "",
        }
      }
      accommodationProcess={[Function]}
      cities={Array []}
      citiesProcess={[Function]}
    >
      <div>
        <form
          id="createAccommodationForm"
          onSubmit={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Promise {},
                },
              ],
            }
          }
          testdata="createAccommodationForm"
        >
          <div
            className="form-content"
          >
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12 m6"
              >
                <input
                  className="validate"
                  disabled={false}
                  id="name"
                  name="name"
                  onChange={[Function]}
                  required={true}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="name"
                >
                  Name
                </label>
              </div>
              <div
                className="input-field col s12 m6"
                id="onwerField"
              >
                <input
                  className="validate"
                  disabled={false}
                  id="owner"
                  name="owner"
                  onChange={[Function]}
                  required={true}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="owner"
                >
                  Owner
                </label>
              </div>
            </div>
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12 m6"
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <select
                  className="validate"
                  disabled={false}
                  name="location"
                  onChange={[Function]}
                  required={true}
                >
                  <option
                    defaultValue="none"
                  >
                    Choose Location
                  </option>
                </select>
                <label
                  htmlFor="location"
                >
                  Location
                </label>
              </div>
              <div
                className="input-field col s12 m6"
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <select
                  className="validate"
                  disabled={false}
                  name="type"
                  onChange={[Function]}
                  required={true}
                >
                  <option
                    defaultValue="none"
                  >
                    Choose type
                  </option>
                  <option
                    value="Mansion"
                  >
                    Mansion
                  </option>
                  <option
                    value="Hotel"
                  >
                    Hotel
                  </option>
                  <option
                    value="Guest House"
                  >
                    Guest House
                  </option>
                </select>
                <label
                  htmlFor="type"
                >
                  Type
                </label>
              </div>
            </div>
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12 m6"
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <select
                  className="validate"
                  disabled={false}
                  name="status"
                  onChange={[Function]}
                >
                  <option
                    defaultValue="none"
                  >
                    Choose status
                  </option>
                  <option
                    value="available"
                  >
                    Available
                  </option>
                  <option
                    value="booked"
                  >
                    Booked
                  </option>
                </select>
                <label
                  htmlFor="status"
                >
                  Status
                </label>
              </div>
              <div
                className="input-field col s12 m6"
                style={
                  Object {
                    "marginTop": "20px",
                  }
                }
              >
                <input
                  className="validate"
                  disabled={false}
                  id="description"
                  name="description"
                  onChange={[Function]}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="description"
                >
                  Additional Info
                </label>
              </div>
            </div>
            <div
              className="row services"
            >
              <h6
                className="headings"
              >
                Services
              </h6>
              <div
                className="row card service"
                id="textInput"
                key="0"
              >
                <div
                  className="input-field col s12 m4"
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  <input
                    className="validate"
                    disabled={false}
                    id="serviceName0"
                    name="name"
                    onChange={[Function]}
                    required={true}
                    testdata="inputChange"
                    type="text"
                  />
                  <label
                    htmlFor="serviceName0"
                  >
                    Sevice Name
                  </label>
                </div>
                <div
                  className="input-field col s12 m4"
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  <input
                    className="validate"
                    disabled={false}
                    id="serviceCost0"
                    name="cost"
                    onChange={[Function]}
                    testdata="inputChange"
                    type="text"
                  />
                  <label
                    htmlFor="serviceCost0"
                  >
                    Sevice Cost
                  </label>
                </div>
                <div
                  className="input-field col s12 m4"
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  <input
                    className="validate"
                    disabled={false}
                    id="serviceDetails0"
                    name="description"
                    onChange={[Function]}
                    testdata="inputChange"
                    type="text"
                  />
                  <label
                    htmlFor="serviceDetails0"
                  >
                    Sevice Details
                  </label>
                </div>
              </div>
              <AddButton
                add={[Function]}
                item="services"
              >
                <button
                  className="btn-floating waves-effect waves-light tooltipped\\" add-btn right"
                  data-tooltip="Add item"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="material-icons"
                  >
                    add
                  </i>
                </button>
              </AddButton>
            </div>
            <div
              className="row card accImages"
              id="textInput"
            >
              <h6
                className="headings"
              >
                Accommodation Images
              </h6>
              <div
                className="row"
                key="0"
              >
                <div
                  className="file-field input-field col m6 s12"
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <div
                    className="btn"
                  >
                    <input
                      accept="image/*"
                      onChange={[Function]}
                      required={true}
                      type="file"
                    />
                    <i
                      className="material-icons"
                    >
                      folder
                    </i>
                  </div>
                  <div
                    className="file-path-wrapper"
                  >
                    <input
                      className="file-path validate"
                      placeholder="Upload image"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="input-field col s12 m6"
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <input
                    className="validate"
                    disabled={false}
                    id="images0"
                    name="details"
                    onChange={[Function]}
                    testdata="inputChange"
                    type="text"
                  />
                  <label
                    htmlFor="images0"
                  >
                    Image Caption
                  </label>
                </div>
              </div>
              <AddButton
                add={[Function]}
                item="images"
              >
                <button
                  className="btn-floating waves-effect waves-light tooltipped\\" add-btn right"
                  data-tooltip="Add item"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="material-icons"
                  >
                    add
                  </i>
                </button>
              </AddButton>
            </div>
            <div
              className="row rooms"
              id="textInput"
            >
              <h6
                className="headings"
              >
                Rooms
              </h6>
              <div
                className="row card room"
                key="0"
              >
                <div
                  className="col s12 m8 image-part"
                >
                  <div
                    className="file-field input-field"
                  >
                    <div
                      className="btn"
                    >
                      <input
                        accept="image/*"
                        onChange={[Function]}
                        required={true}
                        type="file"
                      />
                      <i
                        className="material-icons"
                      >
                        folder
                      </i>
                    </div>
                    <div
                      className="file-path-wrapper"
                    >
                      <input
                        className="file-path validate"
                        placeholder="Upload image"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="input-field col s12 "
                    style={
                      Object {
                        "marginTop": 30,
                      }
                    }
                  >
                    <input
                      className="validate"
                      disabled={false}
                      id="image0"
                      name="details"
                      onChange={[Function]}
                      testdata="inputChange"
                      type="text"
                    />
                    <label
                      htmlFor="image0"
                    >
                      Image Caption
                    </label>
                  </div>
                </div>
                <div
                  className="col s12 m4"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12 "
                      style={
                        Object {
                          "marginTop": "20px",
                        }
                      }
                    >
                      <input
                        className="validate"
                        disabled={false}
                        id="name0"
                        name="name"
                        onChange={[Function]}
                        required={true}
                        testdata="inputChange"
                        type="text"
                      />
                      <label
                        htmlFor="name0"
                      >
                        Room Name
                      </label>
                    </div>
                    <div
                      className="input-field col s12 "
                      style={
                        Object {
                          "marginTop": "20px",
                        }
                      }
                    >
                      <input
                        className="validate"
                        disabled={false}
                        id="roomNumber0"
                        name="roomNumber"
                        onChange={[Function]}
                        required={true}
                        testdata="inputChange"
                        type="text"
                      />
                      <label
                        htmlFor="roomNumber0"
                      >
                        Room Number
                      </label>
                    </div>
                    <div
                      className="input-field col s12"
                      style={
                        Object {
                          "marginTop": "20px",
                        }
                      }
                    >
                      <select
                        className="validate"
                        disabled={false}
                        id="type0"
                        name="type"
                        onChange={[Function]}
                        required={true}
                      >
                        <option
                          defaultValue="none"
                        >
                          Choose type
                        </option>
                        <option
                          value="vip"
                        >
                          VIP
                        </option>
                        <option
                          value="medium"
                        >
                          Medium
                        </option>
                        <option
                          value="low level"
                        >
                          Low level
                        </option>
                      </select>
                      <label
                        htmlFor="type0"
                      >
                        Type
                      </label>
                    </div>
                    <div
                      className="input-field col s12 "
                      style={
                        Object {
                          "marginTop": "20px",
                        }
                      }
                    >
                      <input
                        className="validate"
                        disabled={false}
                        id="cost0"
                        name="cost"
                        onChange={[Function]}
                        testdata="inputChange"
                        type="text"
                      />
                      <label
                        htmlFor="cost0"
                      >
                        Room Cost
                      </label>
                    </div>
                    <div
                      className="input-field col s12 "
                      style={
                        Object {
                          "marginTop": "20px",
                        }
                      }
                    >
                      <input
                        className="validate"
                        disabled={false}
                        id="similarRooms0"
                        name="similarRooms"
                        onChange={[Function]}
                        required={true}
                        testdata="inputChange"
                        type="number"
                      />
                      <label
                        htmlFor="similarRooms0"
                      >
                        Similar Rooms
                      </label>
                    </div>
                    <div
                      className="input-field col s12"
                      style={
                        Object {
                          "marginTop": "20px",
                        }
                      }
                    >
                      <select
                        className="validate"
                        disabled={false}
                        id="roomStatus0"
                        name="status"
                        onChange={[Function]}
                      >
                        <option
                          defaultValue="none"
                        >
                          Choose status
                        </option>
                        <option
                          value="available"
                        >
                          Available
                        </option>
                        <option
                          value="booked"
                        >
                          Booked
                        </option>
                      </select>
                      <label
                        htmlFor="roomStatus0"
                      >
                        Status
                      </label>
                    </div>
                  </div>
                </div>
              </div>
              <AddButton
                add={[Function]}
                item="rooms"
              >
                <button
                  className="btn-floating waves-effect waves-light tooltipped\\" add-btn right"
                  data-tooltip="Add item"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="material-icons"
                  >
                    add
                  </i>
                </button>
              </AddButton>
            </div>
            <div
              className="contentFooter"
            >
              <ToastContainer
                autoClose={8000}
                closeButton={[Function]}
                closeOnClick={true}
                draggable={false}
                draggablePercent={80}
                hideProgressBar={true}
                newestOnTop={false}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-center"
                role="alert"
                rtl={false}
                transition={[Function]}
              >
                <div
                  className="Toastify"
                />
              </ToastContainer>
            </div>
          </div>
          <div
            className="form-footer"
          >
            <div
              className="submitBtn"
            >
              <button
                className="btn waves-effect waves-light"
                disabled={true}
                id="submit"
                type="submit"
              >
                Send Request
              </button>
            </div>
          </div>
        </form>
      </div>
    </AccommodationForm>
  </Connect(AccommodationForm)>
</Provider>
`;

exports[`Test <create accommodation/> Component Should render component correctly 1`] = `
<div>
  <NavBar />
  <main
    className="createAccomContainer"
  >
    <div
      className="sideMenu"
    >
      <div
        className="fixed-sidebar container center hide-on-med-and-down"
      >
        <_default />
      </div>
      <div
        className="show-on-med-and-down container center"
      >
        <mobileSidebar />
      </div>
    </div>
    <div
      className="mainContent"
    >
      <div
        className="row"
      >
        <div
          className="col s12 m2"
        />
        <div
          className="col s12 m10 content"
        >
          <h5
            className="center"
          >
            Create Accommodation
          </h5>
          <Connect(AccommodationForm) />
        </div>
      </div>
    </div>
  </main>
  <_default />
</div>
`;
